app/src/main/java/org/sopt/sample/presentation/signup/SignUpViewModel.kt

```diff
@@ -30,19 +42,25 @@ class SignUpViewModel : ViewModel() {
                 response: Response<ResponseSignup>
             ) {
                 if (response.isSuccessful) {
-                    _signUpResult.value = AuthNetworkState.Success
+                    _signUpResult.value = NetworkState.Success
                 } else {
-                    _signUpResult.value = AuthNetworkState.Failure
+                    _signUpResult.value = NetworkState.Failure
                 }
             }
 
             override fun onFailure(call: Call<ResponseSignup>, t: Throwable) {
-                _signUpResult.value = AuthNetworkState.Error(t)
+                _signUpResult.value = NetworkState.Error(t)
                 Log.e("SIGNUP FAIL", "mes : " + t.message)
 
             }
         })
     }
 
+    private fun emailRegex(email: String): Boolean {
+        return email.matches("^(?=.*[a-z])(?=.*[0-9])[0-9A-Za-z$&+?@#<>^*%!]{6,10}$".toRegex())
+    }
 
+    private fun passwordRegex(password: String): Boolean {
+        return password.matches("^(?=.*[a-z])(?=.*[0-9])(?=.*[$&+?@#<>^*%!])[0-9A-Za-z$&+?@#<>^*%!]{6,12}$".toRegex())
+    }
```

Comment:
```suggestion
    private fun emailRegex(email: String): Boolean {
        return email.matches("^(?=.*[a-zA-Z])(?=.*[0-9]).{6,10}$".toRegex())
    }

    private fun passwordRegex(password: String): Boolean {
        return password.matches("^(?=.*[a-zA-Z])(?=.*[0-9])(?=.*[$&+?@#<>^*%!]).{6,12}$".toRegex())
    }
```
`.`을 쓰면 좀 더 간단하게 정규식을 표현할 수 있는 것 같아요!
대문자도 영문 조건에 포함해주는 건 어떤가요?